packages/mobile/src/firebase/saga.ts

```diff
@@ -178,8 +182,25 @@ function* updatePaymentRequestStatus({ id, status }: UpdatePaymentRequestStatusA
   }
 }
 
+function* declinePaymentRequest({ id }: DeclinePaymentRequestAction) {
+  CeloAnalytics.track(CustomEventNames.incoming_request_payment_decline)
+  yield call(updatePaymentRequestStatus, id, PaymentRequestStatus.DECLINED)
+}
+
+function* completePaymentRequest({ id }: CompletePaymentRequestAction) {
+  CeloAnalytics.track(CustomEventNames.incoming_request_payment_pay)
+  yield call(updatePaymentRequestStatus, id, PaymentRequestStatus.COMPLETED)
+}
+
+function* cancelPaymentRequest({ id }: CancelPaymentRequestAction) {
+  CeloAnalytics.track(CustomEventNames.outgoing_request_payment_cancel)
+  yield call(updatePaymentRequestStatus, id, PaymentRequestStatus.CANCELLED)
+}
+
 export function* watchPaymentRequestStatusUpdates() {
-  yield takeLeading(Actions.PAYMENT_REQUEST_UPDATE_STATUS, updatePaymentRequestStatus)
+  yield takeLeading(Actions.PAYMENT_REQUEST_COMPLETE, completePaymentRequest)
```

Comment:
The downside of this approach is that we are spawning 3x as many saga forks. I don't know if there's a perf impact to spawning saga forks, maybe it's negligible.